### ISSUE TITLE
DataDescription.data_level needs a validator for valid tags

### ISSUE THREAD

**Issue by Sun-flow** [AllenNeuralDynamics/aind-data-schema#695]
**Is your feature request related to a problem? Please describe.**
Users are currently still uploading data which includes outdated `data_level` tags, which need to be then updated manually (via a script that is already written) to our current tag standards.

**Describe the solution you'd like**
Introduce a validator to `DataDescription` that checks whether the tag input is valid or not, and converts it to a valid analog if not. If there is no valid analog, reject the model.

**Describe alternatives you've considered**
We could also solve this via running the `data_description_upgrade` over all `DataDescription`s regularly, but this requires a lot of oversight.

**Comment by jtyoung84** [comment]
The current version of aind-data-schema uses enums, which validate the inputs. A few thoughts:

 - People use older versions of aind-data-schema. We can convince people to upgrade, but there will always be a lag between when aind-data-schema is updated and users upgrade their dependencies
 - I think the overall idea of using a semi-structured Document Store vs. rigid relational databases, was to allow users some flexibility in describing the metadata. @dyf can add some extra context.
 - There were early discussions on whether every schema update should require a corresponding script that migrates the previous version to the current version. I think we settled on just writing upgrade scripts as needed. I'm happy to revisit the discussion though.

**Comment by dyf** [comment]
I agree with @jtyoung84, and I think it makes sense to centralize this logic in the upgrader classes.

**Comment by jtyoung84** [comment]
@jtyoung84 Convert this to user story
